### ISSUE TITLE
Rule proposal: `filename`, the successor of `filename-case`

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2593]
### Description

Deprecate `unicorn/filename-case` and implement the enhanced rule `unicorn/filename`.

Config: 
```
"unicorn/filename": [
  "error",
  {
    "regex": "^[a-z]+\\\\.[tj]sx$",
    "ignore": [".test\\\\.[tj]sx$"]
  }
]
```

`unicorn/filename` allow us control the filename more granularly. For example, we will be able to ban filename like `*.MD` while allow `*.md`.

### Examples

```
"unicorn/filename": [
  "error",
  {
    "regex": "^[a-z]+\\\\.[tj]sx?$",
    "ignore": [".test\\\\.[tj]sx?$"]
  }
]


✅ foo.js
✅ foo.ts
❌ Foo.js
✅ Foo.test.js
```

### Proposed rule name

filename

### Additional Info

The default regex is the regex of `kebabCase` case.

**Comment by sindresorhus** [comment]
I don't see why this requires a new rule.

There is literally no reason to uppercase the extension, so we can just ban it without an option in `filename-case`.

**Comment by sindresorhus** [comment]
`filename-case` already enforces lowercase extensions (since #1336), so `*.MD` is flagged while `*.md` is allowed. A generic regex rule isn't a fit here — that's covered by [`eslint-plugin-check-file`](https://github.com/dukeluo/eslint-plugin-check-file). Closing.
